DEVELOPMENT.md

```diff
@@ -1,24 +1,117 @@
+On a conceptual level, we maintain three categories of branches: 
+* The **deployment branches** version software is was/is/shoud be running on testnet and mainnet. 
+  The naming convention is `v0.minor` and loosely follows semantic versioning. The major version is currently pinned to `0`,
+  as the Flow protocol is not yet fully implemented. The minor version increases at every Height-Coordinated-Upgrade [HCU] or spork (points of breaking downwards compatibility).
+  In a nutshell, you can think of the minor as a version number for the overall protocol. Each time there is a change in the protocol, such that nodes
+  running the old and new version can't be mixed, the minor version increases and we create a new deployment branch
+  (usually from `master`).
+* `master` accumulates all revisions and features that are either fully downwards compatible or can be rolled out via an HCU. Probably the majority of PRs
+  can be merged directly to master or via a feature branch. Furthermore, our ability to roll out upgrades to the live network via HCUs will significantly increase over time.     
+* For most **feature branches**, the protocol changes they implement will hopefully be small enough to be deployed via an HCU (and hence can be merged to `master`).
+  Though, for the foreseeable future, large and broadly-breaking improvements will be necessary for a fast evolution and feature-completion of Flow.
+  Such features should be kept in their individual feature branches.  
+* The `spork` branch, will be created in preparation of a new spork from the most recent `master`. Feature branches holding HCU-**in**compatible upgrades will be
+  consolidated on this `spork` branch. The spork branch is intended to be very short-lived and only exist for a few weeks right before the testnet spork.
+  When the Flow network has committed to the spork on a governance level (at the moment, largely coordinated by the Flow Foundation), the `spork` branch will be
+  merged back to `master`.
+
+We are purposefully continuing the usage of the 'spork' terminology here, to describe a "severely breaking change".
+Specifically, spork describes a deployment of new node software, which requires the entire network to be stopped and rebooted.
+We only carry over a snapshot of the execution state but fully re-initialize the protocol state. Thereby, major upgrades of the protocol,
+incl. migrations of the entire execution state to a new formats become reasonably straight forward.
+Additional context on HCU vs sporks can be found [here](https://developers.flow.com/networks/node-ops/node-operation/hcu#hcu-versus-spork).   
+
+<!---
+TEMPORARY to show the image, will be replaced with commented-out version below before merge 
+-->
+
+<img src='https://github.com/onflow/flow-go/blob/ff8882b2fcef5b7dfa86ac97cde407cc7b9ae3d5/docs/flow-go_branching-convention.png' width='350'>
+
+<!---
+<img src='https://github.com/onflow/flow-go/blob/master/docs/BranchingConvention.png' width='350'>
+-->
```

Comment:
```suggestion

<img src='./docs/flow-go_branching-convention.png' width='350'>

```

I believe both GitHub and most IDEs will interpret relative paths, so we can link to whatever file is at this path in the current branch.